packages/babel-parser/src/parser/expression.js

```diff
@@ -309,7 +309,10 @@ export default class ExpressionParser extends LValParser {
 
       if (this.match(tt.eq)) {
         node.left = this.toAssignable(left, /* isLHS */ true);
-        refExpressionErrors.doubleProto = -1; // reset because double __proto__ is valid in assignment expression
+
+        if (refExpressionErrors.doubleProto >= node.left.start) {
```

Comment:
We can replace `node.left.start` by `startPos`, saving unnecessary double property access.

Reply:
Okay, I will do the same.